compiler/src/main/java/com/dylibso/chicory/compiler/internal/WasmAnalyzer.java

```diff
@@ -264,6 +375,66 @@ public List<CompilerInstruction> analyze(int funcId) {
         return result;
     }
 
+    private static void analyzeTryCatchEnd(
+            List<CompilerInstruction> result, TryCatchBlock tryCatchBlock) {
+
+        // Mark the end of the try block
+        result.add(new CompilerInstruction(CompilerOpCode.LABEL, tryCatchBlock.end));
+
+        // Jump over the exception handler if since no exception was thrown
+        result.add(new CompilerInstruction(CompilerOpCode.GOTO, tryCatchBlock.after));
+
+        // Mark the start of the exception handler
+        result.add(new CompilerInstruction(CompilerOpCode.LABEL, tryCatchBlock.handler));
+
+        // store the exception in a temporary slot
+        result.add(new CompilerInstruction(CompilerOpCode.CATCH_START));
+
+        for (int i = 0; i < tryCatchBlock.ins.catches().size(); i++) {
+            var catchCondition = tryCatchBlock.ins.catches().get(i);
+            long afterCatchLabel = tryCatchBlock.afterCatch[i];
+
+            switch (catchCondition.opcode()) {
+                case CATCH:
+                    result.add(
+                            new CompilerInstruction(
+                                    CompilerOpCode.CATCH_COMPARE_TAG, catchCondition.tag()));
+                    result.add(new CompilerInstruction(CompilerOpCode.IFEQ, afterCatchLabel));
+                    result.add(
+                            new CompilerInstruction(
+                                    CompilerOpCode.CATCH_UNBOX_PARAMS, catchCondition.tag()));
```

Comment:
a slight nit, but it looks like the sequence is always:

- CATCH_COMPARE_TAG tag
- IFEQ label
- CATCH_UNBOX_PARAMS tag

so that makes me wonder if it is necessary for them to be separate opcodes -- but it's almost a matter of taste